File: 0409-Longest-Palindrome/0409.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> counter = new HashMap<>();
        for (char c : s.toCharArray()) {
            counter.put(c, counter.getOrDefault(c, 0) + 1);
        }

        boolean oddFound = false;
        int ans = 0;

        for (int count : counter.values()) {
            if (count % 2 == 0) {
                ans += count;
            } else {
                ans += count - 1;
                oddFound = true;
            }
        }

        if (oddFound) {
            ans += 1;
        }

        return ans;
    }
}

File: 0409-Longest-Palindrome/0409.py
from collections import Counter


class Solution:
    def longestPalindrome(self, s: str) -> int:
        map = Counter(s)
        found_odd = False
        ans = 0

        for c, count in map.items():
            # Add even counts entirely
            if count % 2 == 0:
                ans += count
            else:
                ans += count - 1 # Add the largest even part
                found_odd = True 
        
        # Add one more character for the center if an odd frequency is found
        if found_odd:
            ans += 1
    
        return ans